vault/ha.go

```diff
@@ -721,6 +736,23 @@ func (c *Core) waitForLeadership(newLeaderCh chan func(), manualStepDownCh, stop
 	}
 }
 
+func (c *Core) promoteToPerfStandby(ctx context.Context) error {
+	if c.Sealed() {
+		return fmt.Errorf("stand-by still sealed")
+	}
+	dummyCancel := func() {}
+	if err := c.postUnseal(ctx, dummyCancel, readonlyUnsealStrategy{}); err != nil {
+		return err
+	}
+
+	c.perfStandby = true
+	c.standby = false
```

Comment:
I think we need a lock here, don't we?

```suggestion
	c.stateLock.Lock()
	c.perfStandby = true
	c.standby = false
	c.stateLock.Unlock()
```